src/webgpu/web_platform/canvas/getPreferredCanvasFormat.spec.ts

```diff
@@ -13,6 +13,11 @@ g.test('value')
     Ensure getPreferredCanvasFormat returns one of the valid values.
     `
   )
+  .beforeAllSubcases(t => {
+    if (typeof navigator === 'undefined') {
+      throw new SkipTestCase('navigator does not exist in this environment');
```

Comment:
maybe `skipIf` should be moved to `SubcaseBatchState` from `GPUTestSubcaseBatchState` in another PR.

Reply:
SGTM